libcxx/test/std/utilities/smartptr/adapt/inout_ptr/inout_ptr.verify.cpp

```diff
@@ -26,7 +26,7 @@ int main(int, char**) {
 
     // expected-error-re@*:* {{static assertion failed due to requirement {{.*}}std::shared_ptr<> is not supported}}
     std::ignore = std::inout_ptr(sPtr);
-    // expected-error@*:* {{no matching conversion for functional-style cast from 'std::shared_ptr<int>' to 'std::inout_ptr_t<shared_ptr<int>, _Ptr>' (aka 'inout_ptr_t<std::shared_ptr<int>, int *>'}}
+    // expected-error-re@*:* {{no matching conversion for functional-style cast from 'std::shared_ptr<int>' to 'std::inout_ptr_t<{{(std::)?}}shared_ptr<int>, _Ptr>'{{( \(aka 'inout_ptr_t<std::shared_ptr<int>, int *>')?}}}}
```

Comment:
Can you explain these differences?  I don't really get why this is happening but only 'sometimes' instead of always?

Reply:
It is because libcxx is tested against stable clang in addition to trunk, so we have to make sure the test expects both the **before** and the **after** clang diagnostic outputs.